components/mediation-monitor/message-relay/org.wso2.carbon.relay/src/main/java/org/wso2/carbon/relay/ExpandingMessageFormatter.java

```diff
@@ -182,19 +179,20 @@ private void findAndWrite2OutputStream(MessageContext messageContext,
                 DataHandler dh = (DataHandler) binaryDataNode.getDataHandler();
 
                 DataSource dataSource = dh.getDataSource();
-                if (((StreamingOnRequestDataSource) dataSource).isConsumed()) {
-                    Object httpMethodObj = messageContext.getProperty(Constants.Configuration.HTTP_METHOD);
-                    if ((httpMethodObj instanceof String)
-                        && "POST".equals(httpMethodObj)) {
-                        log.warn("Attempting to send an already consumed request ["
-                                 + messageContext.getTo().getAddress()
-                                 + " POST/Empty Message Body]");
+                if (dataSource instanceof StreamingOnRequestDataSource) {
+                    if (((StreamingOnRequestDataSource) dataSource).isConsumed()) {
+                        Object httpMethodObj = messageContext.getProperty(Constants.Configuration.HTTP_METHOD);
+                        if ((httpMethodObj instanceof String) && "POST".equals(httpMethodObj)) {
```

Comment:
"POST" can be retrieved from a Constant. If this is http action, then we can get it from there.

Reply:
Fixed in https://github.com/wso2/carbon-mediation/pull/863